### ISSUE TITLE
How to close a file?

### ISSUE THREAD

**Issue by luxl** [BlueBrain/HighFive#252]
Why `HighFive::File` has no method `close()`? Should I call `H5Fclose(file.getId())` to close it?
Update:
   I called `H5Fclose(file.getId())`, then try `HighFive::File ("xxx", HighFive::File::ReadWrite | HighFive::File::Create | HighFive::File::Truncate)` to reopen it, but failed with meesage 

    HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:
    #000: c:\autotest\hdf5110-stdrelease-dist-10vs15\build\hdfsrc\src\h5f.c line 444 in H5Fcreate(): unable to create file
      major: File accessibilty
      minor: Unable to open file
    #001: c:\autotest\hdf5110-stdrelease-dist-10vs15\build\hdfsrc\src\h5fint.c line 1531 in H5F_open(): unable to truncate
      a file which is already open
      major: File accessibilty
      minor: Unable to open file

**Comment by tdegeus** [comment]
I think that the file is automatically closed when `HighFive::File` goes out of scope. I.e. normally you should not have to worry about this at all, except that you should not open the same file until the 'old' instance goes out-of-scope. Also, I don't think you should mix HighFive with the HDF5 C interface.

Do you have a minimal example (without the `H5Fclose(file.getId())`) where it did not work for you? Maybe we can help to fix the scope issue.

**Comment by alkino** [comment]
Yes, the whole HighFive is built to work with scope, make the file out of scope and it should be destroyed.
